src/wrapper_template.h

```diff
@@ -72,4 +73,105 @@ inline nb::object wrap_function(Func func, const nb::object& input) {
     throw nb::type_error("Input must be a float, int, list, or 0-D/1-D NumPy array.");
 }
 
+inline nb::object prepare_array_argument(const nb::object &scalar, std::size_t shape){
+    std::vector<double> results(shape);
+    for(std::size_t i = 0; i < shape; i++){
+        results[i] = nb::cast<double>(scalar);
+    }
+    auto result_array = nb::ndarray<double, nb::numpy>(results.data(), {static_cast<std::size_t>(shape)}).cast();
+    return result_array;
+}
+
+inline nb::object wrap_multiargument_function(const MultiargumentFunc &func, const std::vector<nb::object> &input) {
+
+    // Check for scalar types (float or int)
+    bool scalars_only = true;
+    size_t len = 0;
+    for (const auto& argument: input) {
+        if (nb::isinstance<nb::list>(argument)) {
+            auto list = nb::cast<nb::list>(argument);
+            len = nb::len(list);
```

Comment:
I don't like variable name `len` : it resembles python `len` function (https://docs.python.org/3/library/functions.html#len) or `nb::len` method.
Use a different name.